src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -133,6 +133,15 @@ public void addTag(Tag p) {
         tags.add(p);
     }
 
+    /**
+     * Deletes a {@code tag} from the address book.
+     * Deletes the tag from contacts using it as well.
+     */
+    public void deleteTag(Tag tag) {
```

Comment:
Maybe the naming could be better, `tag` to `tagToBeDeleted`?

Reply:
Yep I agree! Thanks for spotting.